src/main/java/gyro/aws/ec2/TransitGatewayResource.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.aws.ec2;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import gyro.aws.AwsResource;
+import gyro.aws.Copyable;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.Wait;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.scope.State;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.ec2.model.CreateTransitGatewayRequest;
+import software.amazon.awssdk.services.ec2.model.CreateTransitGatewayResponse;
+import software.amazon.awssdk.services.ec2.model.DeleteTransitGatewayRequest;
+import software.amazon.awssdk.services.ec2.model.DescribeTransitGatewaysResponse;
+import software.amazon.awssdk.services.ec2.model.Ec2Exception;
+import software.amazon.awssdk.services.ec2.model.TransitGateway;
+import software.amazon.awssdk.services.ec2.model.TransitGatewayState;
+
+/**
+ * Creates a transit gateway.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     aws::transit-gateway transit-gateway-example
+ *        description: "example transit gateway"
+ *        transit-gateway-options
+ *            amazon-side-asn: 64512
+ *            dns-support: enable
+ *            vpn-ecmp-support: enable
+ *            default-route-table-association: disable
+ *            default-route-table-propagation: enable
+ *            auto-accept-shared-attachments: disable
+ *        end
+ *
+ *        tags: {
+ *            Name: "Aws EC2 Transit Gateway Resource Example"
+ *        }
+ *     end
+ */
+@Type("transit-gateway")
+public class TransitGatewayResource extends Ec2TaggableResource<TransitGateway> implements Copyable<TransitGateway> {
+
+    private String description;
+    private TransitGatewayResourceOptions transitGatewayOptions;
```

Comment:
Rename examples everywhere as well
```suggestion
    private TransitGatewayResourceOptions options;
```